src/runtime/test/schema-tests.ts

```diff
@@ -146,16 +148,16 @@ describe('schema', () => {
     assert.throws(() => { new Product({shipDays: '2'}); }, TypeError, 'Type mismatch setting field shipDays');
 
     const product = new Product({});
-    assert.throws(() => { product.name = 6; }, TypeError, 'Type mismatch setting field name');
-    assert.throws(() => { product.url = ['url']; }, TypeError, 'Type mismatch setting field url');
-    assert.throws(() => { product.shipDays = {two: 2}; }, TypeError, 'Type mismatch setting field shipDays');
-    assert.throws(() => { product.isReal = 1; }, TypeError, 'Type mismatch setting field isReal');
+    assert.throws(() => product.mutate(p => p['name'] = 6), TypeError, 'Type mismatch setting field name');
+    assert.throws(() => product.mutate(p => p['url'] = ['url']), TypeError, 'Type mismatch setting field url');
+    assert.throws(() => product.mutate(p => p['shipDays'] = {two: 2}), TypeError, 'Type mismatch setting field shipDays');
+    assert.throws(() => product.mutate(p => p['isReal'] = 1), TypeError, 'Type mismatch setting field isReal');
```

Comment:
The bracket notation `p['field']` is pretty cumbersome - I presume it's a type problem? Is this a temporary thing?

Reply:
Yeah, it's a type problem. The function is annotated like so: `mutationFn: (data: {}) => void` so accessing any properties on it causes a compile error (unless you use quotes like I've done here).

The other options were to type the `data` variable as an `Entity`, which has the same problem (because the properties are actually defined on the Entity subclass `Product`), or to use `EntityInterface`, which has `[index: string]: any;` defined on it. I don't really see the point of using that though, because it just allows you to access any property you like (much like the quoting trick).

Also, both those options are kind of an abuse of the type system, because we aren't passing an actual Entity object to the mutationFn, we're passing a raw Object. The ideal outcome would be if we could tell Typescript what properties that object has on it, but I don't know of any way to achieve that...

Reply:
As discussed offline, let's add an EntityMutator interface with the `[index: string]: any` workaround to allow `p.field` notation in the mutator functions and to document the intended use.

Reply:
Done, called it MutableEntityData -- lmk if you don't like that name, I can change it in a follow up PR